### ISSUE TITLE
detaching track instances at each frm are ok on training?

### ISSUE THREAD

**Issue by hjk9984** [TRI-ML/PF-Track#42]
Hi, first of all, thanks for your amazing work.

I'm trying to train this model, but I keep running into GPU memory issues. So I'm trying to run loss.backward() after each frame. However, it seems like grad_fn remains in track_instance.

In the code from this repo, it looks like track_instance is reset after all frames are runed, so grad_fn doesn't remain.

To fix this, I'm thinking of detaching the attributes in track_instance that have grad_fn after backward() and then passing them along. Would that be okay?

Or, if you know of a better way to handle this, I'd really appreciate it if you could let me know!

**Comment by ziqipang** [comment]
@hjk9984 That would be OK to train, but I am unsure whether the performance would be the same.

If GPU memory is the issue, perhaps freezing the backbone would also work.

Since I wrote the code in 2022, and it is 2025 now, there are also some new tools to save GPU memory, such as flash-attention. Please feel free to give it a try.

**Comment by hjk9984** [comment]
Thanks for your answer

I'm currently trying to train the model while detaching, but running into an issue where the model only outpus the 'vehicle' class.
Have you ever encountered a problem liek this before??

**Comment by ziqipang** [comment]
No, I haven't. I suggest you conducting an overfitting test.
